modules/generator/storage/instance.go

```diff
@@ -50,26 +50,29 @@ type storageImpl struct {
 	overrides Overrides
 	closeCh   chan struct{}
 
-	logger log.Logger
+	logger *slog.Logger
 }
 
 var _ Storage = (*storageImpl)(nil)
 
 // New creates a metrics WAL that remote writes its data.
-func New(cfg *Config, o Overrides, tenant string, reg prometheus.Registerer, logger log.Logger) (Storage, error) {
-	logger = log.With(logger, "tenant", tenant)
+// TODO the passed logger does not include any other context attribute
+// Should we standarize slog and deprecate go-kit/log too?
+func New(cfg *Config, o Overrides, tenant string, reg prometheus.Registerer, _ log.Logger) (Storage, error) {
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil)).With("tenant", tenant)
```

Comment:
With this we lose some logger functionality, like adding the logging line.

This is how Thanos has adapted go-kit/log to slog, maybe we can do the same: https://github.com/SungJin1212/thanos/blob/c280c664de7e7ba591ca0f46908b80a5039b1bc3/pkg/logutil/logutil.go

Reply:
Not sure if adding a new dependency to deprecate an existing one is a wise choice 

`sloggk "github.com/tjhop/slog-gokit"`

Reply:
I can do a similar method to get the log level from the go-kit/log and include the contextual information for this PR. Eventually, we should pass the logger from upstream code

Reply:
>Not sure if adding a new dependency to deprecate an existing one is a wise choice

We're not deprecating one right now, no?

Reply:
I've added the ability to include the source (line of code and method name). I think it's ok for this PR